jni/lib/src/third_party/jni_bindings_generated.dart

```diff
@@ -1489,1615 +1526,2193 @@ class JNINativeInterface extends ffi.Struct {
+      SetIntArrayRegion;
+
+  external ffi.Pointer<
+          ffi.NativeFunction<
+              ffi.Void Function(JLongArray, JSize, JSize, ffi.Pointer<JLong>)>>
+      SetLongArrayRegion;
+
+  external ffi.Pointer<
+          ffi.NativeFunction<
+              ffi.Void Function(
+                  JFloatArray, JSize, JSize, ffi.Pointer<JFloat>)>>
+      SetFloatArrayRegion;
+
+  external ffi.Pointer<
+          ffi.NativeFunction<
+              ffi.Void Function(
+                  JDoubleArray, JSize, JSize, ffi.Pointer<JDouble>)>>
+      SetDoubleArrayRegion;
+
+  external ffi.Pointer<
+          ffi.NativeFunction<
+              JInt Function(JClass, ffi.Pointer<JNINativeMethod>, JInt)>>
+      RegisterNatives;
+
+  external ffi.Pointer<ffi.NativeFunction<JInt Function(JClass)>>
+      UnregisterNatives;
+
+  external ffi.Pointer<ffi.NativeFunction<JInt Function(JObject)>> MonitorEnter;
+
+  external ffi.Pointer<ffi.NativeFunction<JInt Function(JObject)>> MonitorExit;
+
+  external ffi.Pointer<
+          ffi.NativeFunction<JInt Function(ffi.Pointer<ffi.Pointer<JavaVM>>)>>
+      GetJavaVM;
+
+  external ffi.Pointer<
+          ffi.NativeFunction<
+              ffi.Void Function(JString, JSize, JSize, ffi.Pointer<JChar>)>>
+      GetStringRegion;
+
+  external ffi.Pointer<
+          ffi.NativeFunction<
+              ffi.Void Function(JString, JSize, JSize, ffi.Pointer<ffi.Char>)>>
+      GetStringUTFRegion;
+
+  external ffi.Pointer<
+          ffi.NativeFunction<
+              ffi.Pointer<ffi.Void> Function(JArray, ffi.Pointer<JBoolean>)>>
+      GetPrimitiveArrayCritical;
+
+  external ffi.Pointer<
+          ffi.NativeFunction<
+              ffi.Void Function(JArray, ffi.Pointer<ffi.Void>, JInt)>>
+      ReleasePrimitiveArrayCritical;
+
+  external ffi.Pointer<
+          ffi.NativeFunction<
+              ffi.Pointer<JChar> Function(JString, ffi.Pointer<JBoolean>)>>
+      GetStringCritical;
+
+  external ffi.Pointer<
+          ffi.NativeFunction<ffi.Void Function(JString, ffi.Pointer<JChar>)>>
+      ReleaseStringCritical;
+
+  external ffi.Pointer<ffi.NativeFunction<JWeak Function(JObject)>>
+      NewWeakGlobalRef;
+
+  external ffi.Pointer<ffi.NativeFunction<ffi.Void Function(JWeak)>>
+      DeleteWeakGlobalRef;
+
+  external ffi.Pointer<ffi.NativeFunction<JBoolean Function()>> ExceptionCheck;
+
+  external ffi.Pointer<
+          ffi.NativeFunction<JObject Function(ffi.Pointer<ffi.Void>, JLong)>>
+      NewDirectByteBuffer;
+
+  external ffi
+          .Pointer<ffi.NativeFunction<ffi.Pointer<ffi.Void> Function(JObject)>>
+      GetDirectBufferAddress;
+
+  external ffi.Pointer<ffi.NativeFunction<JLong Function(JObject)>>
+      GetDirectBufferCapacity;
+
+  external ffi.Pointer<ffi.NativeFunction<ffi.Int32 Function(JObject)>>
+      GetObjectRefType;
+}
+
+extension GlobalJniEnvExtension on ffi.Pointer<GlobalJniEnv> {
+  @pragma('vm:prefer-inline')
```

Comment:
The VM should be smart enough to make its own inlining decisions. Unless you have measured this to have a performance impact, please remove. (If you have measured this to make a performance impact, file a bug with a reproduction and we should fix it.)